stream/metrics.go

```diff
@@ -40,11 +39,9 @@ func (s *Client) initMetrics() {
 		Name:      "publishCounter",
 		Help:      "Number of publications per function",
 	}, []string{"function"})
-}
 
-func (s *Client) AttachMetrics(m *metrics.Metrics) {
-	m.Register(s.m.RecvCounter)
-	m.Register(s.m.Timing)
-	m.Register(s.m.PublishSize)
-	m.Register(s.m.PublishCounter)
+	s.Metrics.Register(s.m.RecvCounter)
```

Comment:
AttachMetrics it's a pattern used everywhere, if you wanna change it, let's talk about that and do that in separate PR everywhere, so it would not stand out

Reply:
The idea was to avoid multi-step initialization.  I think this is fine since it's not introducing a diverging API to the end-user; it's all abstracted away and the user sees none of it.